packages/payment-processor/src/payment/single-request-proxy.ts

```diff
@@ -0,0 +1,184 @@
+import { Contract, Signer, ethers } from 'ethers';
+import {
+  PaymentReferenceCalculator,
+  getPaymentNetworkExtension,
+} from '@requestnetwork/payment-detection';
+import { ClientTypes, ExtensionTypes, CurrencyTypes } from '@requestnetwork/types';
+import { singleRequestProxyFactoryArtifact } from '@requestnetwork/smart-contracts';
+import { IERC20__factory } from '@requestnetwork/smart-contracts/types';
+
+/**
+ * Deploys a Single Request Proxy contract for a given request.
+ *
+ * @param request - The request data object containing payment network and currency information.
+ * @param signer - The Ethereum signer used to deploy the contract.
+ * @returns A Promise that resolves to the address of the deployed Single Request Proxy contract.
+ * @throws {Error} If the payment network is unsupported, payment chain is not found, payee is not found, or if there are invalid payment network values.
+ *
+ * @remarks
+ * This function supports deploying proxies for ERC20_FEE_PROXY_CONTRACT and ETH_FEE_PROXY_CONTRACT payment networks.
+ * It uses the SingleRequestProxyFactory contract to create either an ERC20 or Ethereum Single Request Proxy.
+ * The function calculates the payment reference and handles the deployment transaction, including waiting for confirmation.
+ * The factory address is automatically determined based on the payment chain using the singleRequestProxyFactoryArtifact.
+ */
+export async function deploySingleRequestProxy(
+  request: ClientTypes.IRequestData,
+  signer: Signer,
+): Promise<string> {
+  const requestPaymentNetwork = getPaymentNetworkExtension(request);
+
+  // Check if the payment network is supported, only ERC20_FEE_PROXY_CONTRACT and ETH_FEE_PROXY_CONTRACT are supported
+  if (
+    !requestPaymentNetwork ||
+    (requestPaymentNetwork.id !== ExtensionTypes.PAYMENT_NETWORK_ID.ERC20_FEE_PROXY_CONTRACT &&
+      requestPaymentNetwork.id !== ExtensionTypes.PAYMENT_NETWORK_ID.ETH_FEE_PROXY_CONTRACT)
+  ) {
+    throw new Error('Unsupported payment network');
+  }
+
+  const paymentChain = request.currencyInfo.network;
+  if (!paymentChain) {
+    throw new Error('Payment chain not found');
+  }
+  // Use artifact's default address for the payment chain
+  const singleRequestProxyFactory = singleRequestProxyFactoryArtifact.connect(
+    paymentChain as CurrencyTypes.EvmChainName,
+    signer,
+  );
```

Comment:
I agree with the coderabbitai comment. 

I also think we should use `createInferface()` and `encodeFunctionData()` like the other payment-processor functions do.